src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/leftcurly/InputLeftCurlyCommentBeforeLeftCurly.java

```diff
@@ -0,0 +1,135 @@
+
+package com.puppycrawl.tools.checkstyle.checks.blocks.leftcurly;
+
+import com.puppycrawl.tools.checkstyle.checks.javadoc.JavadocTag;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+public class InputLeftCurlyCommentBeforeLeftCurly {
+
+    void method1 ()
+        /* violation not reported */ { } // violation ''{' at column 34 should be on the previous line'
+
+    void method2 ()
+    { } // violation ''{' at column 5 should be on the previous line'
+
+    void method3() {
+        if(4>2)
+            /* I am emoji 🤣  */ { // violation ''{' at column 33 should be on the previous line'
+        }
+    }
+
+    InputLeftCurlyCommentBeforeLeftCurly()
+        /* I am comment */ { // violation ''{' at column 24 should be on the previous line'
+        System.out.println("Hello CheckStyle");
+    }
+
+    InputLeftCurlyCommentBeforeLeftCurly(String name)
+    { // violation ''{' at column 5 should be on the previous line'
+        System.out.println("I am int" + name);
+    }
+
+    InputLeftCurlyCommentBeforeLeftCurly(int a) { // ok
+    }
+
+    private void some() throws Exception {
+        try {
+            throw new IOException();
+        } catch (IOException | NullPointerException | ArithmeticException ex) //ok
+            /* warn */ { // violation ''{' at column 21 should be on the previous line'
+        }
+    }
+
+    private void some2() throws Exception {
+        try {
+            throw new IOException();
+        } catch (IOException | NullPointerException | ArithmeticException ex) { //ok
+        }
+    }
+
+    public void multipleBlock()
+        /*CheckStyle*/ /* Checkstyle*/ { // violation ''{' at column 36 should be on the previous line'
+    }
+
+    private class Node {
+        int value;
+        Node next;
+
+        public
+            /*comment*/ Node(int value) /*comment*/ { // ok
+            this.value=value;
+        }
+
+        public Node(int value, Node next)
+            /* comment */ { // violation ''{' at column 23 should be on the previous line'
+            this.value = value;
+            this.next = next;
+        }
+
+    }
+
+    public void array() {
+        Integer[] array = null;
+        /*50*/ for/*51*/ (/*52*/Integer/*53*/ i/*54*/:/*55*/ array/*56*/)/*57*/ {/*58*/ // ok
+            /*59*/
+        }/*60*/
+    }
+}
+
+class checks
+        implements check
+        /* CheckStyle */ { // violation ''{' at column 26 should be on the previous line'
+
+    public static Set<String> processJavadocTag(JavadocTag tag) {
+        final Set<String> references = new HashSet<>();
+        final String identifier = tag.getFirstArg().trim();
+        for (Pattern pattern : new Pattern[]
+                {Pattern.compile(".*"), Pattern.compile(".*")}) {
+
+        }
+        return references;
+    }
+
+    private void method() { // ok
+
+        String a = "🧐🧐🧐";
+
+        String b = "🧐🧐🧐ccvb";
+        if (b.equals("🧐🧐")) { // ok
```

Comment:
I have updated the Pr all the expectation includes https://github.com/checkstyle/checkstyle/pull/12382#issuecomment-1315814760 meet but One test case is failing over here because of emoji,  regex is considering this emoji as in a single column but it is actually looks like `if (b.equals(\"\uD83E\uDDD0\uD83E\uDDD0\")) { // ok"`

Reply:
emoji in comment would not make any effect

Reply:
Please add some test cases like:
```
class C {
    void method1 ()
        /* 😂🥳 */ { } 
}
```
And double check column number of  `{`

Reply:
https://github.com/checkstyle/checkstyle/pull/12382#discussion_r1026045766 it is done